subsys/fs/shell.c

```diff
@@ -334,6 +334,66 @@ static int cmd_read(const struct shell *shell, size_t argc, char **argv)
 	return 0;
 }
 
+static int cmd_dump(const struct shell *shell, size_t argc, char **argv)
```

Comment:
Don't like the "dump" name, could it be "cat"; the "cat" name is quite common among other systems while "dump" is very often used with device operation, flash dump, memory dump, tcp dump.
Add function documentation here; your command takes parameter that needs to be figured out from code, this is not good. Some documentation for this shell function is required.
I know that currently shell lacks documentation and that would be possibly first entry, but if you at least provide doxygen for the function it will be one less function got go into to figure out the doc.